scripts/validate

```diff
@@ -9,14 +9,12 @@ PACKAGES="$(find . -name '*.go' | grep -Ev '.pb.go' | xargs -I{} dirname {} | \
 	sort -u | grep -Ev '(.git|.trash-cache|vendor|bin|k8s/pkg|client)')"
 
 echo Packages: ${PACKAGES}
+
 echo Running: go vet
 go vet ${PACKAGES}
-echo Running: golint
-for i in ${PACKAGES}; do
-    if [ -n "$(golint $i | grep -v 'should have comment.*or be unexported' | grep -v 'just return error instead.' | tee /dev/stderr)" ]; then
-        failed=true
-    fi
-done
-test -z "$failed"
+
+echo Running: golangci-lint run --new-from-rev=HEAD~
+golangci-lint run --new-from-rev=HEAD~
```

Comment:
`HEAD~` means the last commit, but each PR could have multiple commits merged. This will be better to use `--new-from-patch`. If unable to do that, just `golanglint-ci run`, good enough for us.

Reply:
I think `HEAD~` works well with drone CI. Drone always merges the PR to master, so the last commit in CI is merged commit which contains all PR changes.

I created a temporary PR https://github.com/longhorn/longhorn-manager/pull/2408 to test CI. The error lint is not in last commit, but CI still can catch the error.

For local development, `HEAD~` also works. We can check with following steps:

```sh
> git clone https://github.com/FrankYang0529/longhorn-manager -b LH-7366-tmp
> make validate
Running validation
Packages: . ./api ./app ./constant ./controller ./controller/monitor ./csi ./csi/crypto ./datastore ./engineapi ./manager ./meta ./metrics_collector ./metrics_collector/registry ./metrics_collector/workqueue ./recovery_backend ./recovery_backend/backend ./recovery_backend/server ./scheduler ./types ./upgrade ./upgrade/util ./upgrade/v15xto160 ./upgrade/v1beta1 ./util ./util/daemon ./util/server ./webhook ./webhook/admission ./webhook/common ./webhook/conversion ./webhook/error ./webhook/resources/backingimage ./webhook/resources/backingimagedatasource ./webhook/resources/backingimagemanager ./webhook/resources/backup ./webhook/resources/backuptarget ./webhook/resources/backupvolume ./webhook/resources/engine ./webhook/resources/engineimage ./webhook/resources/node ./webhook/resources/orphan ./webhook/resources/recurringjob ./webhook/resources/replica ./webhook/resources/setting ./webhook/resources/sharemanager ./webhook/resources/snapshot ./webhook/resources/supportbundle ./webhook/resources/systembackup ./webhook/resources/systemrestore ./webhook/resources/volume ./webhook/resources/volumeattachment ./webhook/server
Running: go vet
Running: golangci-lint run --new-from-rev=HEAD~
client/schemas.go:13:2: field `nonUsedField` is unused (unused)
	nonUsedField string
	^
FATA[0062] exit status 1
make: *** [validate] Error 1
```

Reply:
I just pulled your PR and added an extra commit, but it couldn't catch it. This should be expected, as Drone should not do any magic here. I know you are worried about running `golangci-lint run` directly seems too aggressive to catch all issues from day 1. 

Have you tried to leverage DRONE env variables to download `https://github.com/longhorn/longhorn-manager/pull/<patch number>.patch`? If it's feasible, then you can use `--new-from-patch`.

```console
➜  longhorn-manager git:(fork/LH-7366-tmp) git log --oneline | head -n 4
dfef1ebb non used field 2
4c0c5a9d add meaningless comment
68c6db8e add error field
e0a31985 feat: replace golint with golangci-lint
➜  longhorn-manager git:(fork/LH-7366-tmp) golangci-lint run --new-from-rev=HEAD~ 
client/schemas.go:12:2: field `nonUsedFiled2` is unused (unused)
        nonUsedFiled2 string
        ^
➜  longhorn-manager git:(fork/LH-7366-tmp) golangci-lint run --new-from-rev=HEAD~2
client/schemas.go:12:2: field `nonUsedFiled2` is unused (unused)
        nonUsedFiled2 string
        ^
client/schemas.go:14:2: field `nonUsedField` is unused (unused)
        nonUsedField string
        ^
```

Reply:
Probably, we have different golangci-lint version? I tried to add another unused field. From my local, golangci-lint can get it too.

```sh
> golangci-lint version
golangci-lint has version v1.55.2 built with go1.21.3 from (unknown, mod sum: "h1:yllEIsSJ7MtlDBwDJ9IMBkyEUz2fYE0b5B8IUgO1oP8=") on (unknown)
> git log --oneline | head -n 4
1ca5d2e3 add error field2
4c0c5a9d add meaningless comment
68c6db8e add error field
e0a31985 feat: replace golint with golangci-lint
> golangci-lint run --new-from-rev=HEAD~
client/schemas.go:13:2: field `nonUsedField` is unused (unused)
	nonUsedField  string
	^
client/schemas.go:14:2: field `nonUsedField2` is unused (unused)
	nonUsedField2 string
	^
```

Reply:
Its 1.55.2. It's weird.

Reply:
I think we are Ok with this PR first, as it will work for CI. The remaining is just need to support local lint as well.